var/spack/repos/builtin/packages/adaptivecpp/package.py

```diff
@@ -0,0 +1,185 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import json
+import os
+from os import path
+
+from llnl.util import filesystem
+
+from spack.package import *
+
+
+class Adaptivecpp(CMakePackage, ROCmPackage):
+    """AdaptiveCpp is an implementation of SYCL and C++ standard parallelism for CPUs and GPUs
+    from all vendors: The independent, community-driven compiler for C++-based heterogeneous
+    programming models.
+    Lets applications adapt themselves to all the hardware in the system - even at runtime!
+
+    This package has been previously known as hipSYCL and, briefly, Open SYCL.
+    """
+
+    homepage = "https://adaptivecpp.github.io/"
+    url = "https://github.com/AdaptiveCpp/AdaptiveCpp/archive/refs/tags/v24.10.0.tar.gz"
+    git = "https://github.com/AdaptiveCpp/AdaptiveCpp.git"
+
+    provides("sycl")
+
+    license("BSD-2-Clause")
+
+    with default_args(submodules=True):
+        version("stable", branch="stable")
+        version("develop", branch="develop")
+        version("24.10.0", commit="7677cf6eefd8ab46d66168cd07ab042109448124")
+        # AdaptiveCpp only officially supports the latest version
+        # but we don't deprecate everything else just yet
+        version("24.06.0", commit="fc51dae9006d6858fc9c33148cc5f935bb56b075")
+        version("24.02.0", commit="974adc33ea5a35dd8b5be68c7a744b37482b8b64")
+        version("23.10.0", commit="3952b468c9da89edad9dff953cdcab0a3c3bf78c")
+
+    variant("cuda", default=False, description="Enable CUDA backend for SYCL kernels")
+    variant("rocm", default=False, description="Enable ROCM backend for SYCL kernels")
+
+    depends_on("cmake@3.5:", type="build")
+    depends_on("cmake@3.9:", type="build", when="@24.06:")
+    depends_on("cmake@3.10:", type="build", when="@24.10:")
+
+    depends_on("boost@1.67: +filesystem +fiber +context cxxstd=17", when="@23.10:")
+    depends_on("python@3:")
+    depends_on("llvm@8: +clang", when="~cuda")
+    depends_on("llvm@9: +clang", when="+cuda")
+    # https://github.com/spack/spack/issues/45029 and https://github.com/spack/spack/issues/43142
+    conflicts("^gcc@12", when="@23.10.0")
+    # LLVM PTX backend requires cuda7:10.1 (https://tinyurl.com/v82k5qq)
+    depends_on("cuda@9:10.1", when="@23.10.0: +cuda ^llvm@9")
+    depends_on("cuda@9:", when="@23.10.0: +cuda ^llvm@10:")
+
+    conflicts("%gcc@:8", msg="AdaptiveCpp needs proper C++17 support to be built, %gcc is too old")
+    conflicts(
+        "^llvm build_type=Debug",
+        when="+cuda",
+        msg="LLVM debug builds don't work with AdaptiveCpp CUDA backend; for "
+        "further info please refer to: "
+        "https://github.com/AdaptiveCpp/AdaptiveCpp/blob/master/doc/install-cuda.md",
+    )
+    # https://github.com/spack/spack/issues/46681
+    conflicts("^llvm@19", when="@24.02.0:24.06.0")
```

Comment:
also add llvm 20 as conflict see #49506
```suggestion
    conflicts("^llvm@19", when="@24.02.0:24.06.0")
    conflicts("^llvm@20", when="@24.02.0:24.06.0")
```
or maybe we can think on using some kind of llvm support matrix like in #49507.